src/Server/Handler/Request/CallToolHandler.php

```diff
@@ -57,6 +62,9 @@ public function handle(Request $request, SessionInterface $session): Response|Er
 
         $this->logger->debug('Executing tool', ['name' => $toolName, 'arguments' => $arguments]);
 
+        $toolCallRequestEvent = new CallToolRequestEvent($request);
+        $this->eventDispatcher?->dispatch($toolCallRequestEvent);
+
```

Comment:
Can you skip the variable here and for other similar places? 

```suggestion
        $this->eventDispatcher?->dispatch($new CallToolRequestEvent($request));
```